api/envoy/config/core/v3/grpc_service.proto

```diff
@@ -55,6 +56,14 @@ message GrpcService {
     // This limit is applied to individual messages in the streaming response and not the total size of streaming response.
     // Defaults to 0, which means unlimited.
     google.protobuf.UInt32Value max_receive_message_length = 4;
+
+    // This provides gRPC client level control over envoy generated headers.
+    // For ``x-envoy-internal`` and ``x-forwarded-for`` header:
+    // If true, the header will be sent. But it can be overridden by setting the ``send_internal`` or ``send_xff``
```

Comment:
I think drop mention of internal code details from API here, since this is documentation intended for folks configuring Envoy (or other xDS data planes), rather than Envoy developers.

Reply:
Sure. I kept the per stream override since I imagine folks will be confused if their configuration doesn't work.
I can remove that part as well if you prefer.